beacon_node/lighthouse_network/src/peer_manager/peerdb.rs

```diff
@@ -709,6 +709,7 @@ impl<TSpec: EthSpec> PeerDB<TSpec> {
                     }
                     PeerConnectionStatus::Banned { .. } => {
                         error!(self.log, "Accepted a connection from a banned peer"; "peer_id" => %peer_id);
+                        // TODO: check if this happens and report the unban back
```

Comment:
This shouldn't happen. But we should report it back for completeness, I think

Reply:
I checked what needs to be changed to address this and changes are quite a few. There are some things we could fix first to make this easier to fix. Would you mind if we add this as is?